sample/host/src/AuthenticatedLayout.tsx

```diff
@@ -1,10 +1,13 @@
 import { Link, Outlet, useNavigate } from "react-router-dom";
-import type { RenderItemFunction, RenderSectionFunction } from "@squide/react-router";
-import { Suspense, useCallback } from "react";
-import { useEventBusListener, useNavigationItems, useRenderedNavigationItems } from "@squide/react-router";
+import { type ReactElement, Suspense, useCallback } from "react";
+import { useEventBusListener, useNavigationItems, useRenderedNavigationItems, type RenderItemFunction, type RenderSectionFunction, isNavigationLink, NavigationLinkRenderProps, NavigationSectionRenderProps } from "@squide/react-router";
 
 import { sessionManager } from "./session.ts";
 
+type RenderLinkItemFunction = (item: NavigationLinkRenderProps, index: number, level: number) => ReactElement;
```

Comment:
It feels odd that RenderLinkItemFunction and RenderSectionItemFunction needs to be defined here